### ISSUE TITLE
Possible bug when using Multigrid with p::s::Triangulation

### ISSUE THREAD

**Issue by adamqc** [dealii/dealii#14994]
The following code
```C++
#include <deal.II/distributed/shared_tria.h>

#include <deal.II/dofs/dof_handler.h>

#include <deal.II/fe/fe_q.h>

#include <deal.II/grid/grid_generator.h>

int main(int argc, char **argv) {
  dealii::Utilities::MPI::MPI_InitFinalize init(argc, argv, 1);

  dealii::parallel::shared::Triangulation<3> tria(MPI_COMM_WORLD,
      dealii::Triangulation<3>::limit_level_difference_at_vertices);

  // dealii::Triangulation<3> tria(dealii::Triangulation<3>::limit_level_difference_at_vertices);
  dealii::GridGenerator::hyper_cube(tria, -1, 1);

  dealii::DoFHandler<3> dof_handler(tria);

  dealii::FE_Q<3> fe(1);

  dof_handler.distribute_dofs(fe);
  dof_handler.distribute_mg_dofs();

  return 0;
}
```
will run into assertion failure

```
--------------------------------------------------------
An error occurred in line <344> of file <.../source/distributed/shared_tria.cc> in function
    const std::vector<types::subdomain_id> &dealii::parallel::shared::Triangulation<3>::get_true_level_subdomain_ids_of_cells(const unsigned int) const [dim = 3, spacedim = 3]
The violated condition was: 
    level < true_level_subdomain_ids_of_cells.size()
Additional information: 
    This exception -- which is used in many places in the library --
    usually indicates that some condition which the author of the code
    thought must be satisfied at a certain point in an algorithm, is not
    fulfilled. An example would be that the first part of an algorithm
    sorts elements of an array in ascending order, and a second part of
    the algorithm later encounters an element that is not larger than the
    previous one.
    
    There is usually not very much you can do if you encounter such an
    exception since it indicates an error in deal.II, not in your own
    program. Try to come up with the smallest possible program that still
    demonstrates the error and contact the deal.II mailing lists with it
    to obtain help.
```

Any clue?

**Comment by adamqc** [comment]
I found the reason for the assertion failure was due to my forgetting to set the `construct_multigrid_hierarchy` argument in the constructor of `p::s::Triangulation`. 

However, I think we should add an assertion to check whether the `construct_multigrid_hierarchy` was set in an early stage, perhaps in `DoFHandlerImplementation::Policy::ParallelShared::distribute_mg_dofs`. This may help the user to find the problem easier.

I will create a PR to fix it.
